Compile bug: vulkan-shaders-gen is installed into the build location instead of $PREFIX/bin/vulkan-shaders-gen

### Git commit

5664

### Operating systems

BSD

### GGML backends

CPU

### Problem description & steps to reproduce

vulkan-shaders-gen is installed into /usr/ports/misc/llama-cpp/work/.build/bin/vulkan-shaders-gen imstead of /usr/local/bin/vulkan-shaders-gen

### First Bad Commit

_No response_

### Compile command

```shell
n/a
```

### Relevant log output

```shell
n/a
```

I am building the the FreeBSD port directory.
It used to break in the install target, but now it breaks in the build target:

It fails like this in the build target:

[ 50% 1/2] Building CXX object CMakeFiles/vulkan-shaders-gen.dir/vulkan-shaders-gen.cpp.o
[100% 2/2] Linking CXX executable vulkan-shaders-gen
[ 25% 44/175] cd /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/vulkan-shaders-gen-prefix/src/vulkan-shaders-gen-build && /usr/local/bin/cmake --install . --config Release && /usr/local/bin/cmake -E touch /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/vulkan-shaders-gen-prefix/src/vulkan-shaders-gen-stamp/vulkan-shaders-gen-install
-- Installing: /usr/ports/misc/llama-cpp/work/stage/usr/ports/misc/llama-cpp/work/.build/Release/./vulkan-shaders-gen
[ 25% 45/175] cd /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan && /usr/local/bin/cmake -E make_directory /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/CMakeFiles && /usr/local/bin/cmake -E touch /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/CMakeFiles/vulkan-shaders-gen-complete && /usr/local/bin/cmake -E touch /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/vulkan-shaders-gen-prefix/src/vulkan-shaders-gen-stamp/vulkan-shaders-gen-done
[ 26% 46/175] cd /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan && /usr/ports/misc/llama-cpp/work/.build/Release/vulkan-shaders-gen --glslc /usr/local/bin/glslc --input-dir /usr/ports/misc/llama-cpp/work/llama.cpp-b5684/ggml/src/ggml-vulkan/vulkan-shaders --output-dir /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/vulkan-shaders.spv --target-hpp /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/ggml-vulkan-shaders.hpp --target-cpp /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/ggml-vulkan-shaders.cpp --no-clean
FAILED: ggml/src/ggml-vulkan/ggml-vulkan-shaders.hpp ggml/src/ggml-vulkan/ggml-vulkan-shaders.cpp /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/ggml-vulkan-shaders.hpp /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/ggml-vulkan-shaders.cpp 
cd /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan && /usr/ports/misc/llama-cpp/work/.build/Release/vulkan-shaders-gen --glslc /usr/local/bin/glslc --input-dir /usr/ports/misc/llama-cpp/work/llama.cpp-b5684/ggml/src/ggml-vulkan/vulkan-shaders --output-dir /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/vulkan-shaders.spv --target-hpp /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/ggml-vulkan-shaders.hpp --target-cpp /usr/ports/misc/llama-cpp/work/.build/ggml/src/ggml-vulkan/ggml-vulkan-shaders.cpp --no-clean
/bin/sh: /usr/ports/misc/llama-cpp/work/.build/Release/vulkan-shaders-gen: not found
ninja: build stopped: subcommand failed.

--- a/ggml/src/ggml-vulkan/CMakeLists.txt
+++ b/ggml/src/ggml-vulkan/CMakeLists.txt
@@ -144,7 +144,14 @@ if (Vulkan_FOUND)
                    ${VULKAN_SHADER_GEN_CMAKE_ARGS}
 
         BUILD_COMMAND   ${CMAKE_COMMAND} --build   . --config $<CONFIG>
-        INSTALL_COMMAND ${CMAKE_COMMAND} --install . --config $<CONFIG>
+
+        # NOTE: When DESTDIR is set using Makefile generators and
+        # "make install" triggers the build step, vulkan-shaders-gen
+        # would be installed into the DESTDIR prefix, so it is unset
+        # to ensure that does not happen.
+
+        INSTALL_COMMAND ${CMAKE_COMMAND} -E env --unset=DESTDIR
+                        ${CMAKE_COMMAND} --install . --config $<CONFIG>
     )
 
     set (_ggml_vk_host_suffix $<IF:$<STREQUAL:${CMAKE_HOST_SYSTEM_NAME},Windows>,.exe,>)

--- a/ggml/src/ggml-vulkan/CMakeLists.txt
+++ b/ggml/src/ggml-vulkan/CMakeLists.txt
@@ -146,7 +146,6 @@ if (Vulkan_FOUND)
         BUILD_COMMAND   ${CMAKE_COMMAND} --build   . --config $<CONFIG>
         INSTALL_COMMAND ${CMAKE_COMMAND} --install . --config $<CONFIG>
     )
-    ExternalProject_Add_StepTargets(vulkan-shaders-gen build install)
 
     set (_ggml_vk_host_suffix $<IF:$<STREQUAL:${CMAKE_HOST_SYSTEM_NAME},Windows>,.exe,>)
     set (_ggml_vk_genshaders_dir "${CMAKE_BINARY_DIR}/$<CONFIG>")
@@ -172,8 +171,6 @@ if (Vulkan_FOUND)
 
         DEPENDS ${_ggml_vk_shader_files}
                 vulkan-shaders-gen
-                vulkan-shaders-gen-build
-                vulkan-shaders-gen-install
 
         COMMENT "Generate vulkan shaders"
     )

